app/models/miq_filter.rb

```diff
@@ -1,11 +1,25 @@
 module MiqFilter
+  include NetworkManager
+
   def self.belongsto2object(tag)
     belongsto2object_list(tag).last
   end
 
+  def self.find_object_by_special_class(name)
```

Comment:
For both these methods, it would be more generic if you can inspect the name here to determine the class and return an error if it's not one of the supported "special" classes.